### PR TITLE
[SPARK-41859][SQL] CreateHiveTableAsSelectCommand should set the overwrite flag correctly

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

`CreateHiveTableAsSelectCommand` is used in two APIs:
1. SQL CTAS
2. `DataFrameWriter.saveAsTable`

For SQL CTAS, if the table already exists, we either do nothing if `IF NOT EXISTS` is specified, or fail otherwise. This means, `CreateHiveTableAsSelectCommand` from SQL CTAS should only trigger table insertion if table not exists.

For `DataFrameWriter.saveAsTable`, it can create the table if it doesn't exist, or append data to the table if `SaveMode` is `Append`. `DataFrameWriter.saveAsTable` will drop the table first if `SaveMode` is `Overwrite`. This means, `CreateHiveTableAsSelectCommand` from `DataFrameWriter.saveAsTable` either appends data to an existing table, or creates a new table.

In summary, the table insertion command triggered by `CreateHiveTableAsSelectCommand` should always set `overwrite` to false.

### Why are the changes needed?

fix a mistake

### Does this PR introduce _any_ user-facing change?

No. If the table does not exist, the `overwrite` flag doesn't matter. But it could be a potential bug in the future.

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @viirya @ulysses-you

**Comment by ulysses-you** [comment]
lgtm

**Review by viirya — APPROVED** [review]

**Comment by cloud-fan** [comment]
```
python/pyspark/pandas/sql_processor.py:221: error: unused "type: ignore" comment
```
I don't think the failure is related to this PR.

**Comment by cloud-fan** [comment]
merging to master, thanks!

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala [hunk 1]
```diff
@@ -95,7 +94,6 @@ case class CreateHiveTableAsSelectCommand(
   }
 
   private def getWritingCommand(
-      catalog: SessionCatalog,
       tableDesc: CatalogTable,
       tableExists: Boolean): DataWritingCommand = {
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala:98]: nit but seems like `tableExists` isn;t used anymore too.
